src/Services/PaymentService.php

```diff
@@ -104,6 +104,38 @@ public function updateAuthorization($payment, Authorization $authorization): Aut
         return $authorization;
     }
 
+    public function performPreAuthorization(
+        Authorization $authorization,
+                      $paymentType,
+                      $customer = null,
+        Metadata      $metadata = null,
+        Basket        $basket = null
+    ): Authorization
+    {
+        $payment = $this->createPayment($paymentType);
```

Comment:
In general, what exactly do we do here ?

Reply:
We are preparing payment object that hold and connect all relevant payment data for the payment request. This also makes sure that a viable type object or id is used.
But i just realized that this method is not actually used so i will remove it. Reason is that the existing authorization method works just fine with preauth type.

Reply:
removed method. reusing existing one